src/main/resources/application.yml

```diff
@@ -1,22 +1,21 @@
 spring:
+  application:
+    name: ${SPRING_APPLICATION_NAME:@artifactId@}
   config:
     import: optional:file:.env[.properties]
-
+  profiles:
+    default: ${ACTIVE_PROFILES:local}
```

Comment:
i thing default can be just `local`
if we have more application profiles and want to override `local` we can do it with Environment Variable `SPRING_PROFILES_ACTIVE`
more about it here: 
- [Profiles](https://docs.spring.io/spring-boot/reference/features/profiles.html)
- [Properties and Configuration / Set the Default Profile Name](https://docs.spring.io/spring-boot/how-to/properties-and-configuration.html#howto.properties-and-configuration.set-default-spring-profile-name)

Reply:
it should be: 
```yml
spring:
  profiles:
    default: local
```
if we need to override it for a specific Env. we should set `SPRING_PROFILES_ACTIVE` env. variable on that environment and the value for it could any other configured profile.
where "configured profile" is:
- classes annotated with `@Configuration`
- any other external config properties stored in `application-{profile-name}.yml` file.

Reply:
how then to be with Application.yml which we counted by default?

Reply:
application.yaml file is not a profile as such.
profile is an abstract thing. 
you don't need a yaml file to have a profile created

profile is optional. it'll be resolved in case it resource exists. there are 2 possible resources for a profile:
- Configuration class
- yaml file
if spring finds any of them, it tries to resolve and apply them

what is `spring,profiles.default` ? 
this is the property that lets you set some default profile name to be looked for in case if -Dspring.profiles.active VM option is not set and SPRING_PROFILES_ACTIVE env variable not set too.
but, you can create resources for a profile that you want to have as default, let's say "blah" profile.
and you create 2 resources for it: 
- configuration class annotated with `@Profile("blah")`
- yml file `application-blah.yam`
then, you put this property in your "main" application.yml: `spring,profiles.default = "blah"` and that makes your spring boot application to resolve and pick that configuration annotated with `@Profile("blah")` and that `application-blah.yml` if you do not supply -Dspring.profiles.active VM option and you do not set SPRING_PROFILES_ACTIVE env variable and the machine where you run your java application.

Reply:
Corrected.